package/src/components/MessageInput/CooldownTimer.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Text } from 'react-native';
+
+import { BuiltinRoles, ChannelResponse, Role } from 'stream-chat';
+import dayjs from 'dayjs';
+
+import { useChannelContext, useChatContext } from '../../contexts';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+} from '../../types/types';
+
+type Roles = Array<Role>;
+
+const SECOND_IN_MS = 1000;
+
+/**
+ * useCooldown can be used to start a cooldown defined
+ * for a Channel by setting an end time for
+ **/
+export const useCooldown = <
```

Comment:
Can we move these two hooks to separate files under the directory - `components/MessageInput/hooks/`